analysis_configs/sequences_s_channel_scouting.py

```diff
@@ -150,6 +150,40 @@ def add_analysis_branches(events):
     
     return events
 
+def add_dark_quark_matching(events):
+    #performs delta R matching between AK8 jets and dark quarks and saves the indices of the matched jets in a new branch
+    #(uses new functions for now, should check if it also works with the old ones)
+
+    jet_eta = events.FatJet_eta
+    jet_phi = events.FatJet_phi
+
+    dark_quark_eta = events.MatrixElementGenParticle_eta
+    dark_quark_phi = events.MatrixElementGenParticle_phi
+
+    # matching with the first dark quark
+    dR1 = event_vars.delta_r_dark_quark(dark_quark_eta[:,0], dark_quark_phi[:,0], jet_eta[:,:], jet_phi[:,:]) <= 0.8
+    dR1 = ak.values_astype(dR1, int)
+    matched_index1 = ak.argmax(dR1, axis=1)
```

Comment:
You are aware that with `argmax` you get the first occurrence of True? So you will match only the highest pT jet with delta R less than 0.8?

Reply:
Yes I am aware of that, this was my way of preventing 3 matched jets in an event
because AK8 jets are not always cone-like, 2 jets can sometimes (very rarely) be matched to the same dark quark
my thought was that we would want to just select the higher pT jet in that case